sdks/python/src/opik/cli/export_experiment.py

```diff
@@ -0,0 +1,235 @@
+"""Experiment export functionality."""
+
+import sys
+from pathlib import Path
+from typing import Optional
+
+import click
+from rich.console import Console
+
+import opik
+from opik.cli.export_utils import (
+    create_experiment_data_structure,
+    debug_print,
+    write_json_data,
+    print_export_summary,
+)
+from opik.cli.export_dataset import export_experiment_datasets
+from opik.cli.export_prompt import (
+    export_experiment_prompts,
+    export_related_prompts_by_name,
+)
+from opik.cli.export_project import export_traces
+
+console = Console()
+
+
+def export_experiment_by_id(
+    client: opik.Opik,
+    output_dir: Path,
+    experiment_id: str,
+    dataset: Optional[str],
```

Comment:
dataset_name property will be available in the Experiment instance returned by `client.get_experiment_by_id()` so I'm not sure we need it here